src/main/c/CMakeLists.txt

```diff
@@ -68,26 +68,35 @@ else()
 endif()
 
 add_library(seasocks_obj OBJECT ${SEASOCKS_SOURCE_FILES})
-target_include_directories(seasocks_obj PUBLIC .)
+target_include_directories(seasocks_obj PUBLIC
+    $<INSTALL_INTERFACE:${CMAKE_INSTALL_INCLUDEDIR}/>
+    $<BUILD_INTERFACE:${CMAKE_CURRENT_SOURCE_DIR}/.>
+)
 set_property(TARGET seasocks_obj PROPERTY POSITION_INDEPENDENT_CODE TRUE)
 
 add_library(seasocks STATIC $<TARGET_OBJECTS:seasocks_obj> $<TARGET_OBJECTS:embedded>)
 add_library(Seasocks::seasocks ALIAS seasocks)
-target_include_directories(seasocks PUBLIC .)
+target_include_directories(seasocks PUBLIC
+    $<INSTALL_INTERFACE:${CMAKE_INSTALL_INCLUDEDIR}/>
+    $<BUILD_INTERFACE:${CMAKE_CURRENT_SOURCE_DIR}/.>
+)
 target_link_libraries(seasocks PRIVATE ${CMAKE_THREAD_LIBS_INIT})
 if (DEFLATE_SUPPORT)
     target_link_libraries(seasocks PRIVATE "${ZLIB_LIBRARIES}")
 endif()
 
 add_library(seasocks_so SHARED $<TARGET_OBJECTS:seasocks_obj> $<TARGET_OBJECTS:embedded>)
 add_library(Seasocks::seasocks_so ALIAS seasocks_so)
-target_include_directories(seasocks_so PUBLIC ${ZLIB_INCLUDE_DIRS} .)
+target_include_directories(seasocks_so PUBLIC ${ZLIB_INCLUDE_DIRS} 
+    $<INSTALL_INTERFACE:${CMAKE_INSTALL_INCLUDEDIR}/>
+    $<BUILD_INTERFACE:${CMAKE_CURRENT_SOURCE_DIR}/.>
```

Comment:
In these lines I had to split out the build and install interface. This ensures that we don't export paths to the build directory, but instead use the installed header files when `Seasocks::seasocks` is used from an installed version of Seasocks.

Reply:
Is there a way to improve these targets in general? Eg. by propagating the include directories by `seasocks_obj`?

Reply:
Hmm, good question! I agree that would be ideal. I naively tried linking against the object library with `target_link_libraries(seasocks PUBLIC seasocks_obj)` to have those include directories propagate. But cmake doesn't allow that for the `OBJECT` library type:

```
  Target "seasocks_obj" of type OBJECT_LIBRARY may not be linked into another
  target.  One may link only to STATIC or SHARED libraries, or to executables
  with the ENABLE_EXPORTS property set.
```
Some searching later; turns out that this [feature](https://cmake.org/cmake/help/v3.12/command/target_link_libraries.html#linking-object-libraries) is added in cmake version [3.12](https://cmake.org/cmake/help/v3.12/release/3.12.html#commands) (last entry of `commands`). Cmake 3.12 is from November 2018, I don't think it's a good idea to require that version for this minor improvement. So I propose to keep this as is for now.

Reply:
> Cmake 3.12 is from November 2018, I don't think it's a good idea to require that version for this minor improvement. So I propose to keep this as is for now.

Yes, that makes sense.